authLdap.php

```diff
@@ -211,15 +221,16 @@ function authLdap_options_panel()
  *
  * throws exception if there is a problem connecting
  *
+ * @param boolean $reset; allows a forceful reset of the LDAP server connections for External User scenario
  * @conf boolean authLDAPDebug true, if debugging should be turned on
  * @conf string  authLDAPURI LDAP server URI
  *
  * @return Org_Heigl\AuthLdap\LdapList LDAP server object
  */
-function authLdap_get_server()
+function authLdap_get_server($reset = false)
 {
 	static $_ldapserver = null;
-	if (is_null($_ldapserver)) {
+	if (is_null($_ldapserver)|| $reset === "reset") {
```

Comment:
When `$reset` is a boolean value this comparison will always be `false`

Reply:
This actually isn't a boolean (I neglected to update the comment), but rather an explicit string of "reset". I should actually set a default value of `null`, or change it to a boolean - whichever will make the code behave better. But this should only trigger when explicitly called by line 363 `$ldapServerList = authLdap_get_server("reset");`.